CS4125Project/src/Services/DetailedReceiptBuilder.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package Services;
+
+import java.util.ArrayList;
+
+/**
+ *
+ * @author Jack
+ */
+public class DetailedReceiptBuilder implements ReceiptBuilder{
+    private Receipt receipt;
+    public DetailedReceiptBuilder(){
+        this.receipt = new Receipt();
+    }
+    
+    public void buildusername(){
+        receipt.SetUsername(Shop.getInstance().getAccount().getUsername());
+    }
+    
+    public void buildEmail(){
+        receipt.SetEmail(Shop.getInstance().getAccount().getEmail());
+    }
+    
+    public void buildPaymentDetails(){
+        receipt.SetPaymentDetails(Shop.getInstance().getAccount().getPaymentType().getPaymentInfo());
+    }
+  
+    public void buildPurchases(){ 
+        ArrayList<Purchase>purchases = ShoppingBasket.GetInstance().GetBasketContents();
+        String itemList = "You have " + purchases.size() + " items.\n";
```

Comment:
This doesn't print the number of items, this prints the number of purchases.
Eg. If I have 2 frogs and 1 apple, I have 3 items but this will print that I only have 2 since it's counting number of purchases.